handlers.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/jinzhu/gorm"
+	"github.com/lib/pq"
+)
+
+// Image is a representation of an image entry.
+type Image struct {
+	CreatedAt time.Time      `json:"created_at"`
+	UpdatedAt time.Time      `json:"updated_at"`
+	ID        string         `json:"id" binding:"required" gorm:"primary_key;type:varchar(64)"`
+	Name      string         `json:"name" binding:"required" form:"name"`
+	Tags      pq.StringArray `json:"tags,omitempty" gorm:"type:varchar(100)[]" form:"tags"`
+}
+
+func getImages(db *gorm.DB) gin.HandlerFunc {
+	fn := func(c *gin.Context) {
+		var image Image
+		var images []Image
+
+		// Return a specific image entry.
```

Comment:
instead of checking conditions on presence of required params I'd extract the block below to separate method.